src/plugins/grpc/client.js

```diff
@@ -0,0 +1,144 @@
+'use strict'
+
+const FORMAT_TEXT_MAP = require('opentracing').FORMAT_TEXT_MAP
+const log = require('../../log')
+const pick = require('lodash.pick')
+const Tags = require('opentracing').Tags
+
+function getFilter (config, filter) {
+  if (typeof config[filter] === 'function') {
+    return config[filter]
+  }
+
+  if (config[filter] instanceof Array) {
+    return element => pick(element, config[filter])
+  }
+
+  if (config.hasOwnProperty(filter)) {
+    log.error(`Expected '${filter}' to be an array or function.`)
+  }
+
+  return null
+}
+
+function getMethodType (definition) {
+  if (definition.requestStream) {
+    if (definition.responseStream) {
+      return 'bidi'
+    }
+
+    return 'client_stream'
+  }
+
+  if (definition.responseStream) {
+    return 'server_stream'
+  }
+
+  return 'unary'
+}
+
+function patch (grpc, tracer, config) {
+  if (config.client === false) return
+
+  config = config.client || config
+
+  const configFields = getFilter(config, 'fields')
+  const configMetadata = getFilter(config, 'metadata')
+
+  function datadogInterceptor (options, nextCall) {
+    const path = options.method_definition.path
+    const methodType = getMethodType(options.method_definition)
+    const scope = tracer.scopeManager().active()
+    const span = tracer.startSpan('grpc.request', {
+      childOf: scope && scope.span(),
+      tags: {
+        [Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_CLIENT,
+        'grpc.method.name': path,
+        'grpc.method.type': methodType,
+        'resource.name': path.split('/').slice(-1),
+        'service.name': config.service || `${tracer._service}-grpc-client`
+      }
+    })
+
+    return new grpc.InterceptingCall(nextCall(options), {
+      sendMessage: (message, next) => {
+        if (configFields) {
+          const values = configFields(message)
+
+          for (const key in values) {
+            span.setTag(`grpc.request.message.fields.${key}`, values[key])
+          }
+        }
+
+        next(message)
+      },
+
+      cancel: () => {
+        span.setTag('grpc.request.status', grpc.status.CANCELLED)
```

Comment:
The tag should probably be `grpc.request.status.code` to allow future extensions to `grpc.request.status`.

Reply:
Just realized that it's also `grpc.request.status.code` for the server. Is that the expected name? Shouldn't it be `response`?

I wonder if just `grpc.status.code` would be simpler since there is a single status code for the entire request on response. This is also what we do for HTTP.